src/api/fileops/fileset.ts

```diff
@@ -78,10 +78,7 @@ export class Fileset<FormDataType extends IFormData<F>, T, D, F> implements IRes
     const sharedUploadingProcess = uploadingProcess.pipe(
       shareReplay()
     );
-
-    sharedUploadingProcess.subscribe();
-
-    return this.watch().pipe(
+    const watcher = this.watch().pipe(
```

Comment:
It'd be great if you move this observable to a separate method :+1:

Reply:
Well, it wouldn't be great, to be honest.  If you look to `watch().pip()` flow you will find out that it is using a lot of vars from the current method scope as well as some of its arguments. Moving this method out will cause a lot of extra coding and I don't like it :)

Reply:
you only use `filesCompleted` and `allFilesCompleted` and none of them would be scope of the `Observable.create` part. this method looks too big, but this change is not required though :+1: